tst/test_tuner.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+import logging
+from pathlib import Path
+
+import numpy as np
+import pytest
+
+from syne_tune import StoppingCriterion
+from syne_tune import Tuner
+from syne_tune.config_space import randint
+from syne_tune.optimizer.baselines import RandomSearch
+from syne_tune.util import script_height_example_path
+from tst.util_test import temporary_local_backend
+
+
+@pytest.fixture
+def tunertools():
+    max_steps = 100
+    sleep_time = 0.02
+    max_wallclock_time = 1
+    mode = "min"
+    metric = "mean_loss"
+
+    config_space = {
+        "steps": max_steps,
+        "sleep_time": sleep_time,
+        "width": randint(0, 20),
+        "height": randint(-100, 100),
+    }
+    entry_point = script_height_example_path()
+    scheduler = RandomSearch(config_space, metric=metric, mode=mode)
+    trial_backend = temporary_local_backend(entry_point=entry_point)
+    stop_criterion = StoppingCriterion(
+        max_wallclock_time=max_wallclock_time, min_metric_value={"mean_loss": -np.inf}
+    )
+    return scheduler, stop_criterion, trial_backend
+
+
+def test_tuner_not_wait_trial_completion_when_stopping(tunertools):
```

Comment:
i know, it's not yet ready for review but have a look at `@pytest.mark.parametrize`.

Reply:
Thanks! Will definitely include in the final version.